drivers/flash/soc_flash_nrf_rram.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_rram_controller
+
+#include <string.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/barrier.h>
+#include <hal/nrf_rramc.h>
+
+#include <zephyr/../../drivers/flash/soc_flash_nrf.h>
+
+LOG_MODULE_REGISTER(flash_nrf_rram, CONFIG_FLASH_LOG_LEVEL);
+
+#define RRAM DT_INST(0, soc_nv_flash)
+
+#define RRAM_START DT_REG_ADDR(RRAM)
+#define RRAM_SIZE  DT_REG_SIZE(RRAM)
+
+#define RRAM_WORD_SIZE 4
+
+#define PAGE_SIZE  DT_PROP(RRAM, erase_block_size)
+#define PAGE_COUNT ((RRAM_SIZE) / (PAGE_SIZE))
+
+#define ERASE_VALUE 0xFF
+
+static struct k_sem sem_lock;
+#define SYNC_INIT()   k_sem_init(&sem_lock, 1, 1)
+#define SYNC_LOCK()   k_sem_take(&sem_lock, K_FOREVER)
+#define SYNC_UNLOCK() k_sem_give(&sem_lock)
+
+#if CONFIG_NRF_RRAM_WRITE_BUFFER_SIZE > 0
+#define WRITE_BUFFER_ENABLE 1
+#define WRITE_BUFFER_SIZE   CONFIG_NRF_RRAM_WRITE_BUFFER_SIZE
+#define WRITE_LINE_SIZE     16 /* In bytes, one line is 128 bits. */
+#else
+#define WRITE_BUFFER_ENABLE 0
+#define WRITE_BUFFER_SIZE   0
+#define WRITE_LINE_SIZE     RRAM_WORD_SIZE
+#endif
+
+#ifndef CONFIG_SOC_FLASH_NRF_RADIO_SYNC_NONE
+#define FLASH_SLOT_WRITE 5000
+
+#if WRITE_BUFFER_ENABLE
+#define RRAM_MAX_WRITE_BUFFER (WRITE_BUFFER_SIZE * WRITE_LINE_SIZE)
+#else
+#define RRAM_MAX_WRITE_BUFFER (WRITE_LINE_SIZE * 32)
```

Comment:
Why?

Reply:
@nvlsianpu Could you explain? This has a strange value to me as well.

Reply:
According to information from @nvlsianpu, he meant the size of the native RRAM buffer (128-bits, not bytes). Changed.